helix-term/src/commands/typed.rs

```diff
@@ -3184,67 +3099,66 @@ pub(super) fn command_mode(cx: &mut Context) {
+
+                TYPABLE_COMMAND_MAP
+                    .get(command)
+                    .map(|tc| tc.completer_for_argument_number(argument_number_of(&shellwords)))
+                    .map_or_else(Vec::new, |completer| {
+                        completer(editor, word)
+                            .into_iter()
+                            .map(|(range, mut file)| {
+                                file.content = shellwords::escape(file.content);
+
+                                // offset ranges to input
+                                let offset = input.len() - len;
+                                let range = (range.start + offset)..;
+                                (range, file)
+                            })
+                            .collect()
+                    })
             }
         }, // completion
         move |cx: &mut compositor::Context, input: &str, event: PromptEvent| {
-            let parts = input.split_whitespace().collect::<Vec<&str>>();
-            if parts.is_empty() {
-                return;
-            }
+            match expand_args(cx.editor, input.into(), true) {
+                Ok(args) => {
+                    let shellwords = Shellwords::from(args.as_ref());
+                    let command = shellwords.command();
 
-            // If command is numeric, interpret as line number and go there.
-            if parts.len() == 1 && parts[0].parse::<usize>().ok().is_some() {
-                if let Err(e) = typed::goto_line_number(cx, &[Cow::from(parts[0])], event) {
-                    cx.editor.set_error(format!("{}", e));
-                }
-                return;
-            }
+                    if command.is_empty() {
+                        return;
+                    }
 
-            // Handle typable commands
-            if let Some(cmd) = typed::TYPABLE_COMMAND_MAP.get(parts[0]) {
-                let shellwords = Shellwords::from(input);
-                let args = shellwords.words();
+                    // If input is `:NUMBER`, interpret as line number and go there.
+                    if command.parse::<usize>().is_ok() {
+                        if let Err(err) = typed::goto_line_number(cx, Args::from(command), event) {
+                            cx.editor.set_error(format!("{err}"));
+                        }
+                        return;
+                    }
 
-                if let Err(e) = (cmd.fun)(cx, &args[1..], event) {
-                    cx.editor.set_error(format!("{}", e));
+                    // Handle typable commands
+                    if let Some(cmd) = typed::TYPABLE_COMMAND_MAP.get(command) {
+                        if let Err(err) = (cmd.fun)(cx, shellwords.args(), event) {
+                            cx.editor.set_error(format!("{err}"));
+                        }
+                    } else if event == PromptEvent::Validate {
+                        cx.editor.set_error(format!("no such command: '{command}'"));
+                    }
```

Comment:
Doesn't there need to be an expansion here?

Reply:
```rust
        match expand_args(cx.editor, input.into(), true) {
                Ok(args) => {
                    let shellwords = Shellwords::from(args.as_ref());
```

shellwords are created after the expansion has been done

Reply:
Ahh, I see it. The diff was a little confusing.